controllers/hcpvaultsecretsapp_controller.go

```diff
@@ -288,6 +299,58 @@ func (r *HCPVaultSecretsAppReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}, nil
 }
 
+func (r *HCPVaultSecretsAppReconciler) startShadowSecretCleanupRoutine(ctx context.Context) {
+	logger := log.FromContext(ctx)
+	// cleanup of orphaned shadow secrets runs once every hour
+	ticker := time.NewTicker(1 * time.Hour)
+
+	// this process is expected to run indefinitely
+	for {
+		select {
+		case <-ticker.C:
+			err := r.cleanupOrphanedShadowSecrets(ctx)
+			if err != nil {
+				logger.Error(err, "Failed to cleanup orphaned shadow secrets")
+			}
+		}
+	}
+}
+
+func (r *HCPVaultSecretsAppReconciler) cleanupOrphanedShadowSecrets(ctx context.Context) error {
```

Comment:
This would be a great function to test with the fake k8s client (look for calls to `testutils.NewFakeClientBuilder()` for examples). It allows you to specify the k8s objects (i.e. HCPVaultSecretsApp's and/or Secret's) that will come back from calls to client.List and client.Get for a given test case.